### PR TITLE
UPAPI-31 - User / Groups / Roles refactor

### PR DESCRIPTION
**Pull request by psabbate**

### CODE REVIEW

File: src/test/java/com/alsterup/Mocks.java [hunk 1]
```diff
@@ -276,7 +287,8 @@ public static UserDTO mockUserDTO(User user) {
         .email(user.getEmail())
         .username(user.getUsername())
         .status(user.getStatus().toString())
-        .roles(user.getRoles().stream().map(Role::getName).collect(Collectors.toSet()))
+        //
+        // .roles(user.getRoles().stream().map(Role::getName).collect(Collectors.toSet()))
```
Comment (tpeirotti) [inline src/test/java/com/alsterup/Mocks.java]: Remove this

File: src/test/java/com/alsterup/Mocks.java [hunk 2]
```diff
@@ -285,7 +297,7 @@ public static RoleDTO mockRoleDTO(Role role) {
         .id(role.getId())
         .name(role.getName())
         .description(role.getDescription())
-        .system(role.isSystemRole())
+        //                .deletable(role.isSystemRole())
```
Comment (tpeirotti) [inline src/test/java/com/alsterup/Mocks.java]: We shouldn't let commented code in master.

File: src/test/java/com/alsterup/Mocks.java [hunk 3]
```diff
@@ -295,7 +307,7 @@ public static RoleSearchDTO mockRoleSearchDTO(Role role) {
         .id(role.getId())
         .name(role.getName())
         .description(role.getDescription())
-        .system(role.isSystemRole())
+        //                .system(role.isSystemRole())
```
Comment (tpeirotti) [inline src/test/java/com/alsterup/Mocks.java]: We shouldn't let commented code in master.

File: src/main/java/com/alsterup/security/model/User.java [hunk 4]
```diff
@@ -21,10 +20,11 @@
     indexes = {
       @Index(columnList = "accountId, verificationToken", name = "verificationToken_hidx")
     })
+@EqualsAndHashCode(callSuper = true)
 public class User extends BaseEntity {
 
   public static final String[] HEADERS = {
-    "id", "firstName", "lastName", "username", "email", "roles", "status"
+    "id", "firstName", "lastName", "username", "email", "status"
```
Comment (tpeirotti) [inline src/main/java/com/alsterup/security/model/User.java]: We need to update the CSVWriterBuilder if we change the headers.

Comment (psabbate) [reply src/main/java/com/alsterup/security/model/User.java]: mmm ok, I would have expected that removing the roles from the headers just avoid writting them. Let me check.

Comment (tpeirotti) [reply src/main/java/com/alsterup/security/model/User.java]: com/alsterup/security/service/impl/UserServiceImpl.java:426

Comment (tpeirotti) [reply src/main/java/com/alsterup/security/model/User.java]: We could add a validation in where if the first written row has a different size than the headers we throw an exception.

**Pull request merged**